### DISCUSSION TITLE
🔍 Setting Default Select Value for Prisma Fields

### DISCUSSION THREAD

**Discussion by Sajarin-M** [prisma/prisma#20346] [Q&A] [answered]
I have certain fields in my Prisma model, such as `createdAt`, `updatedAt`, and `deletedAt`. When querying the database using Prisma, I want to avoid retrieving data for these fields unless explicitly stated in the `select` option. 

❓ **How can I achieve this? Is there a way to set default select values for specific fields?**

🤔 Additionally, is this possible with Prisma extensions?

**Comment by ludralph** [comment, marked as answer]
Hi @Sajarin-M 👋 

To avoid retrieving these fields unless explicitly stated in the select option, you can use Prisma's `select` query modifier. The `select` modifier allows you to specify which fields you want to retrieve from the database. By default, if you do not include the select modifier, Prisma will retrieve all fields defined in the model.
Here's an example of how you can use the `select` modifier to exclude certain fields:
```js
const user = await prisma.user.findUnique({
  where: { id: 1 },
  select: {
    id: true,
    name: true,
    // Add only the fields you want to retrieve explicitly
  },
});
```
In this example, only the id and name fields will be retrieved from the database, and `createdAt`, `updatedAt`, and `deletedAt` will be excluded.

You can also write a Prisma Client extension for this as well. For example:
```js
const selectiveRetrievalExtension = {
  query: {
   user: {
      $allOperations({ args, query, operation }) {
        const { select } = args; // Extract the `select` field from query arguments
        // If `select` is not provided or doesn't include the special fields,
        // exclude the `createdAt`, `updatedAt`, and `deletedAt` fields from the query
        if (!select || (!select.createdAt && !select.updatedAt && !select.deletedAt)) {
          args.select = {
            ...select,
            id: true,
            name: true,
            createdAt: false,
            updatedAt: false,
            deletedAt: false,
          };
        }

        return query(args);
      },
    },
  },
};
```

This extension will only return the `id` and `name` fields. This approach ensures that the `createdAt`, `updatedAt`, and `deletedAt` fields are not retrieved by default unless explicitly requested, 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by Sajarin-M** [reply]
Thank you, @ludralph! :wave: By the way, I'm curious whether this will work well with Typescript ?

**Reply by ludralph** [reply]
I don't forsee a problem with using this with Typescript

**Reply by tutods** [reply]
Currently isn't possible to change the `args.select`, any way to do it now?
